src/VehicleQuery.php

```diff
@@ -259,29 +269,31 @@ public function findAll()
         }
 
         $data = $each['Vehicles'];
-        
-        for ($i=2;$i<=$each['TotalPages'];$i++) {
+
+        for ($i = 2; $i <= $each['TotalPages']; $i++) {
             $query = new self();
             $query->setClient($this->getClient());
             $query->setPage($i);
             $query->where($this->_where);
             $r = $query->getResponse();
-            
+
             $data = array_merge($data, $r['Vehicles']);
         }
         
         return $this->createIterator($data, $each['TotalMatches'], 1, $each['TotalMatches'], 1);
     }
-    
+
     /**
+     * Retrieves a single vehicle by its ID.
      *
-     * @param integer $id The id of the vehicle
-     * @return \Indielab\AutoScout24\Vehicle
+     * @param int $id The unique identifier of the vehicle to retrieve.
+     * @return Vehicle The vehicle object corresponding to the provided ID.
+     * @throws Exception If the response from the client is invalid or an error occurs.
      */
-    public function findOne($id)
+    public function findOne(int $id): Vehicle
     {
-        $response = $this->getClient()->endpointResponse('vehicles/'.$id, $this->_where);
-        
+        $response = $this->getClient()->endpointResponse('vehicles/' . $id);
```

Comment:
@JUVOJustin $this->_where has been removed. why?

Reply:
Good catch! Sorry for the inconvenience.